setup.py

```diff
@@ -177,13 +191,35 @@ def finalize_options(self):
         )
 
     def run(self):
-        pkg_config_version_check('cairo', CAIRO_VERSION_REQUIRED)
+        ext = self.extensions[0]
+        global is_msvc
+
+        if os.name == 'nt':
```

Comment:
What about
```
distutils.ccompiler import get_default_compiler
get_default_comiler() == 'msvc'
```

Reply:
Hi @Lazka,

Hmm, I will look into using that, thought I could not do that.  I think I still need to check whether one passed in --compiler=mingw32 though.

The .pc file, since I am also looking at getting PyGObject to build with Visual Studio out-of-the-box, I could forgo it, unless there are other packages that want pycairo via pkg-config.

For the CI part, I think what might work is that I can provide MSVC 2008 (for Python 2.7), 2010 (for 3.3, 3.4) and 2015 (for 3.5, 3.6) builds of Cairo with their dependencies on my github, and ket AppVeyor download them, unpack them as needed, set INCLUDE, LIB, PATH to find the headers, .lib and .dll's, and run the build and tests against those builds.

Reply:
> Hmm, I will look into using that, thought I could not do that. I think I still need to check whether one passed in --compiler=mingw32 though.

I didn't know about the compiler flag, never mind then. I need to look how build_ext.py figures things out.

> The .pc file, since I am also looking at getting PyGObject to build with Visual Studio out-of-the-box, I could forgo it, unless there are other packages that want pycairo via pkg-config.

Unrelated, but I was recently thinking of also porting PyGObject to distutils, like here with pycairo. Would that help you in any way? Or do you have any thoughts on that potential change?

> For the CI part, I think what might work is that I can provide MSVC 2008 (for Python 2.7), 2010 (for 3.3, 3.4) and 2015 (for 3.5, 3.6) builds of Cairo with their dependencies on my github, and ket AppVeyor download them, unpack them as needed, set INCLUDE, LIB, PATH to find the headers, .lib and .dll's, and run the build and tests against those builds.

Thanks. I've found this in the meantime http://preshing.com/20170529/heres-a-standalone-cairo-dll-for-windows/ which, if I'm not mistaken, should work for all Python versions as it includes the runtime and pycairo doesn't malloc/free anything which could conflict. But I'm new to all this, so if that doesn't work out I'll come back to your offer :)

Reply:
Hi @lazka,

For PyGObject, there were some discussions about whether to use distutils over autotools, and it seems that along the way autotools is preferred (but then, as you probably know, there is an ongoing trend to build things with Meson within GNOME--and that was why this PR initially tries to do a .pc file for this package for Visual Studio build, as Meson frequently uses pkg-config to discover dependencies, which, in GNOME for MSVC builds, we use (which is in a review process) a manual discovery scheme where we try to locate the related header(s) and then find the .lib that is most typically generated with the dependencies' build system that is used for MSVC builds).

It now seems that there are wrapper distutils setup.py that actually calls the autotools build routines, so I actually have a wip/msvc branch in git.gnome.org for PyGObject that basically uses NMake Makefiles to take care of things in a similar fashion on the Visual Studio side (which, actually, forgoes the use of pkg-config in them), which I do rebase and update from time to time.  I do think, however, the patches in there will probably need more refinements to go into PyGObject upstream, and I think there is the issue of Python 3.5/3.6 not working too well on the MSVC builds, along with x64 issues.

My take on this.

With blessings, thank you!